### ISSUE TITLE
Terraform Plan Tab in Pipeline runs details doesn't show plans 

### ISSUE THREAD

**Issue by brampostma** [charleszipp/azure-pipelines-tasks-terraform#342]
**Describe the bug**
I have a Azure Devops Pipeline in yaml with a Terraform Plan task in every Stage (with stages dev,tst,acc,prd). Those Plan tasks publish their plans, this works fine and I can see the plans from the pipeline run overview page. The problem is that when the plan task of the first stage (dev) does not have any changes it will not publish anything to the pipeline run overview, even tho the other stages plan tasks contain have changes.

**To Reproduce**
1. Setup a pipeline with multiple stages and a terraform@cli tasks in each stage:
```  
    - task: TerraformCLI@0
    displayName: 'Terraform : plan'
    inputs:
      command: plan
      publishPlanResults: ${{ parameters.publishPlanResults }}
```
2. Execute pipeline where the first stage doesn't have any changes and further stages do.
3. Look at Terraform plan tab in pipeline run results page.

**Expected behavior**
I expect to see the plans of the other stages to be visable in the pipeline run overview, but i get this:
![image](https://user-images.githubusercontent.com/34541396/236818242-78768a5c-ed02-4acd-b30b-daa55b7a378e.png)

even tho the test stage has changes:
![image](https://user-images.githubusercontent.com/34541396/236819653-f69de39d-80f5-43ac-903e-9a79ac8b4b4e.png)

**Comment by jason-johnson** [comment]
More details in #390

**Comment by BHellstream** [comment]
Yeah, this is pretty annoying :)  and makes the feature almost unusable for multistage (with plan) pipelines

**Comment by tomas0620** [comment]
Same problem, @jason-johnson do you have a fix scheduled for this feature? THX.

**Comment by sam-bowman** [comment]
Just adding a note to say that I'm seeing the same behaviour. If one plan is empty then none at all will show on the terraform plan tab in a multistage pipeline, despite all publishPlanResults field having unique names per stage. As long as all terraform plan outputs have changes, then the published results shows correctly.

**Comment by swisman** [comment]
Any update on this @jason-johnson?
After having played around and I can share that it depends on the terraform version one is using: 
the Terraform Plan tab does not show any plan in a multi stage terraform pipeline if the there are no changes planned for the first stage when for example running on terraform v1.8.0, v1.9.0 and v1.9.7 (latest).
However, it does work as expected if I'm using v0.14.7: both plans (an empty) and a real one are shown in the tab for a two stage pipeline:

![image](https://github.com/user-attachments/assets/bc068a36-d855-4d36-9bb8-bb917eab7d46)
![image](https://github.com/user-attachments/assets/5cfa25cc-2efb-487c-83ef-87f235e697b6)

Looking at the trace reveals the same issue as reported in https://github.com/jason-johnson/azure-pipelines-tasks-terraform/issues/390 - I'm getting a 404 when trying to send a GET request to an url like

https://dev.azure.com/myorg/myproject-guid/_apis/build/builds/13846/707783bc-7f13-4f6e-b5de-a7d88dc36a62/63ac96dc-06b0-5bf8-1a99-855724a4a5b3/attachments/terraform-plan-results/global

if no changes are planned in the first stage and running on a more recent terraform version (see top). In summary what's happening (as others already described): 
if there are no changes planned for the first stage, no plan ca be found for it, the code fails and stops looking for other plans to be displayed in the Terraform Plan tab.

**Comment by jason-johnson** [comment]
@brampostma can you check now using `terraformCli@2`?  This is the unstable version which currently has a potential fix to this issue (#467)

**Comment by jason-johnson** [comment]
@BHellstream @sam-bowman @swisman @tomas0620 Can anyone check if this is still happening with `@2`?  I am unable to reproduce this and I think the issue may have been name clashes which shouldn't happen anymore.

I set up a test with a stage which just makes an empty plan file followed by a bunch of plans, some of which having non-empty plans.

Also keep in mind that the "No plans have been published" is like the loading screen.  That will always appear until some plans are loaded and sometimes the loading does take a few seconds.

**Comment by tomas0620** [comment]
Hi,

I don't know what I changed after my previous comment, but the plan tab has been working until now. 
However, yesterday I noticed the plan was not showing up ( also on old builds where I am sure it was working).

I found in the dev tools/ network that the request to Timeline(https://.../_apis/build/builds/.../Timeline) ends with a 404 response.
This request is from "azdo-attachment-service.ts:68"

When I tried to call it from Postman with all the headers, it was not working. I found that the problem is in the header "accept". It contains "api-version=7.2-preview.3;" 
When I change this to 7.1 the request is working also from Postman.

**Comment by BHellstream** [comment]
Tried it out, example of the pipeline conf:
  
      - task: TerraformCLI@2
        displayName: 'TF plan management'
        inputs:
          command: 'plan'
          workingDirectory: $(mgmtDir)
          environmentServiceName: $(rootServiceConnection)
          providerAzureRmSubscriptionId: $(backendSubID)
          commandOptions: '-out mgmt.tfplan -detailed-exitcode'
          publishPlanResults: 'plan-management'`

And here is the graphical stage view, the bottom one that is just null the top one is reported as nothing to change which is correct, but the bottom one has changes and a plan is created.
![image](https://github.com/user-attachments/assets/3ccf9e9f-7bd3-434e-b298-1ebfbd571bb5)
          
But still nothing: 
![image](https://github.com/user-attachments/assets/88053562-077c-4c91-a9ad-f0075942fc69)

I did get the terraform tab to click on though, that was missing altogether for some reason when i use @1 now, it has been there before...

**Comment by BHellstream** [comment]
Ok, interesting, i did a follow up since i realized that my use case there with two parallel stages is a bit different from series, this is a bigger picture of the same example i had in the previous comment: 
![image](https://github.com/user-attachments/assets/433a5fdd-587c-4eda-a71d-a1b785b7be61)

now there is no changes in the first top one, none in the bottom one, but a change in the second on the top, and that is getting a plan:
![image](https://github.com/user-attachments/assets/1296a9ed-82e0-4764-b6fe-6a366a09f025)

**Comment by jason-johnson** [comment]
Ok, the fixes I've added only apply to `@2` right now.  I have a multi-stage and multi-job which has one stage publishing an empty plan and one stage with 2 jobs publishing non-empty plans.  Everything shows up as expected but it takes about 3 seconds to load the plan.  Right now I'm finishing up adding a loading message so we can tell if the extension thinks there are no plans or if the load is failing (what I suspect).

**Comment by tomas0620** [comment]
@jason-johnson  I have still a problem to load Timeline by [azdo-attachment-service:68](https://github.com/jason-johnson/azure-pipelines-tasks-terraform/blob/main/views/terraform-plan/src/services/attachments/azdo-attachment-service.ts#L68) (see my previous comment)
Do you think this timeline is needed? The code does not use the returned value.

The version of my server is -  Azure DevOps Server 2022.2 (AzureDevopsServer_20240806.7)

**Comment by jason-johnson** [comment]
Looking at it, it seems like that function pulls a lot of things that aren't actually used.  There may have been plans here, before my time, to show more information or perhaps something that was shown before isn't anymore but I will clean up the function a bit and make a new `@2` release in the next few hours.

**Comment by jason-johnson** [comment]
Ok, @tomas0620 can you check now?  `@2` no longer uses the timeline stuff.

**Comment by jason-johnson** [comment]
@BHellstream @tomas0620 @swisman @brampostma @sam-bowman Can you check if you still have the issue with version `@2` of the task?  This version will be published to `@1` as soon as I'm sure the issue is resolved.

**Comment by BHellstream** [comment]
Not working for me, it's stuck in loading plans:
![image](https://github.com/user-attachments/assets/7b3eec17-3622-49c4-bb43-53204dd37c5b) 

new error though, now i get a 404 when it tries to fetch the plan: 
![image](https://github.com/user-attachments/assets/1b1b6456-c6fd-4088-aff3-ab9fb6111d89)

Two stages that run in parallel, the bottom one has changes top one don't. 
![image](https://github.com/user-attachments/assets/88cca52b-ca18-4124-8c25-77a71cdd238d)

**Comment by BHellstream** [comment]
But now i did a second test where both stages produce a plan and i get the same result there, it's stuck in loading plans.
same 404 error.

**Comment by BHellstream** [comment]
I now get 404 in all scenarios where it tries to fetch a plan. 

version details
![image](https://github.com/user-attachments/assets/3cde5d86-5b62-4c27-984f-87e5742586de)

**Comment by jason-johnson** [comment]
> Not working for me, it's stuck in loading plans: ![image](https://private-user-images.githubusercontent.com/16521030/393503780-7b3eec17-3622-49c4-bb43-53204dd37c5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDM3ODAtN2IzZWVjMTctMzYyMi00OWM0LWJiNDMtNTMyMDRkZDM3YzViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2Y2JlMmE2ZDgwNTlkNjdkNTNlY2M3NTUxNDg5YjliZTM5ZGJlZjE0YzVhZmUwMDk2MTg0YWU5Mjg1ZjFmNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FyQOecQrsUrFBUYQqBhnZi3F3wdVcIGyWFj_MGhnGOo)
> 
> new error though, now i get a 404 when it tries to fetch the plan: ![image](https://private-user-images.githubusercontent.com/16521030/393503510-1b1b6456-c6fd-4088-aff3-ab9fb6111d89.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDM1MTAtMWIxYjY0NTYtYzZmZC00MDg4LWFmZjMtYWI5ZmI2MTExZDg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1ZDNhMWU1YTJjYjc2MzcxYzMyZDM0MGYyZmM1OGRlNTYxY2M0MTEwMmM0Y2ZjMjAzMDA1YjFlY2NlYjlhNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UNT5GjiTAE1FesZ1jQlteCQ8Vg-SKbCOGsBTHtSEptc)
> 
> Two stages that run in parallel, the bottom one has changes top one don't. ![image](https://private-user-images.githubusercontent.com/16521030/393503638-88cca52b-ca18-4124-8c25-77a71cdd238d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDM2MzgtODhjY2E1MmItY2ExOC00MTI0LThjMjUtNzdhNzFjZGQyMzhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1NDAxMmIzMDNkZGEyZmJhMzgwZTUyNGI1ZDY0NjI1ZGNmMmJjM2ZkMWYxYmZjODNiODNhMTdjNjc2YzBjODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7eA8r5JDfHmlxMBbNVRQrLsT569h_mEsszTm5mtCeCY)

It was always "stuck" actually.  Just previously the code would show the "no plans exist" or whatever by default and if there were plans they would be loaded.  Now it shows "loading" initially when the plans are not yet loaded and after the load function completes it should update to either the "no plans" message if there are none or the plans should show as normal.

**Comment by jason-johnson** [comment]
> I now get 404 in all scenarios where it tries to fetch a plan.
> 
> version details ![image](https://private-user-images.githubusercontent.com/16521030/393505434-3cde5d86-5b62-4c27-984f-87e5742586de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDU0MzQtM2NkZTVkODYtNWI2Mi00YzI3LTk4NGYtODdlNTc0MjU4NmRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxYjkxZmNlYzE0NTQyZTAzMWVmMmFlMTgxNjNkOTMxOGM3MGNhNjZmODdiYTI2YjE4MTYxZDVkNzgwMjdiZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.f4591vveLwWEIkNXE6I27L3Tuaet9MiSNv4JwHTYP9s)

Are you able to see exactly what URL is getting 404?  I have exactly this scenario and everything is running without error.  Are you running this with a hosted agent, private agent or on prem?

**Comment by BHellstream** [comment]
> > Not working for me, it's stuck in loading plans: ![image](https://private-user-images.githubusercontent.com/16521030/393503780-7b3eec17-3622-49c4-bb43-53204dd37c5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDM3ODAtN2IzZWVjMTctMzYyMi00OWM0LWJiNDMtNTMyMDRkZDM3YzViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2Y2JlMmE2ZDgwNTlkNjdkNTNlY2M3NTUxNDg5YjliZTM5ZGJlZjE0YzVhZmUwMDk2MTg0YWU5Mjg1ZjFmNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FyQOecQrsUrFBUYQqBhnZi3F3wdVcIGyWFj_MGhnGOo)
> > new error though, now i get a 404 when it tries to fetch the plan: ![image](https://private-user-images.githubusercontent.com/16521030/393503510-1b1b6456-c6fd-4088-aff3-ab9fb6111d89.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDM1MTAtMWIxYjY0NTYtYzZmZC00MDg4LWFmZjMtYWI5ZmI2MTExZDg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1ZDNhMWU1YTJjYjc2MzcxYzMyZDM0MGYyZmM1OGRlNTYxY2M0MTEwMmM0Y2ZjMjAzMDA1YjFlY2NlYjlhNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UNT5GjiTAE1FesZ1jQlteCQ8Vg-SKbCOGsBTHtSEptc)
> > Two stages that run in parallel, the bottom one has changes top one don't. ![image](https://private-user-images.githubusercontent.com/16521030/393503638-88cca52b-ca18-4124-8c25-77a71cdd238d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDM2MzgtODhjY2E1MmItY2ExOC00MTI0LThjMjUtNzdhNzFjZGQyMzhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1NDAxMmIzMDNkZGEyZmJhMzgwZTUyNGI1ZDY0NjI1ZGNmMmJjM2ZkMWYxYmZjODNiODNhMTdjNjc2YzBjODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7eA8r5JDfHmlxMBbNVRQrLsT569h_mEsszTm5mtCeCY)
> 
> It was always "stuck" actually. Just previously the code would show the "no plans exist" or whatever by default and if there were plans they would be loaded. Now it shows "loading" initially when the plans are not yet loaded and after the load function completes it should update to either the "no plans" message if there are none or the plans should show as normal.

Ok, but looking at the @1 version from Friday i get no 404 on the plan core in this case the one that has changes, 
Here it fetches each plan successfully, two contains the default message no changes and the core one has the changes and they are displayed as expected. 

![image](https://github.com/user-attachments/assets/2d196fba-5d9f-4207-8c00-d85325ab470b)

now that i re-ran the pipeline still @1 version i now get a 404 I have no idea what might have changed between then and now. 
Looks like @1 and @2 behaves the same now. 
![image](https://github.com/user-attachments/assets/91427c54-c57c-4755-9814-4db061d07dc3)

![image](https://github.com/user-attachments/assets/68a9061c-8357-4358-8c36-f2a00b8633fa)

EDIT:
Well this was not strictly true i see, i re-ran with changes only in core and using @1 and then it worked the same way as on Friday.

**Comment by BHellstream** [comment]
> > I now get 404 in all scenarios where it tries to fetch a plan.
> > version details ![image](https://private-user-images.githubusercontent.com/16521030/393505434-3cde5d86-5b62-4c27-984f-87e5742586de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NzM3NTEsIm5iZiI6MTczMzU3MzQ1MSwicGF0aCI6Ii8xNjUyMTAzMC8zOTM1MDU0MzQtM2NkZTVkODYtNWI2Mi00YzI3LTk4NGYtODdlNTc0MjU4NmRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDEyMTA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxYjkxZmNlYzE0NTQyZTAzMWVmMmFlMTgxNjNkOTMxOGM3MGNhNjZmODdiYTI2YjE4MTYxZDVkNzgwMjdiZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.f4591vveLwWEIkNXE6I27L3Tuaet9MiSNv4JwHTYP9s)
> 
> Are you able to see exactly what URL is getting 404? I have exactly this scenario and everything is running without error. Are you running this with a hosted agent, private agent or on prem?

Yeah, 
`/_apis/build/builds/437826/ee555269-4c79-4d09-9098-21179dea39c2/ddf12597-4c66-5c23-18fd-838ff18747c6/attachments/terraform-plan-results/plan-management `
is working 

`/_apis/build/builds/437798/580546d9-9dfe-436d-8b72-43d36e5974a6/ddf12597-4c66-5c23-18fd-838ff18747c6/attachments/terraform-plan-results/plan-management `
is not working

only difference is the build as far as i can see, i use self hosted agents running on private endpoint in azure, but i get a plan for core in @1 for example so

**Comment by jason-johnson** [comment]
Are your agents up to date?  I can't understand what you're seeing here.  `@1` was updated 2 weeks ago and hasn't been touched since then.  Also I don't understand how there are two calls that have different build IDs.  The way the code works is that it gets the build ID when it start to render that Plans Summary page and then pulls them.  There is no way for that to call 2 different build IDs because it's not checked between plans.

**Comment by BHellstream** [comment]
Ok, i have re-done tests a bit more structured now. 

i use @2, made changes to my core plan, then everything worked as expected, then i added changes to my connectivity, everything still works as expected, i can see two plans 

then i add changes to my parallel stage management, that one gets a 404 and then the GUI is stuck at:
![image](https://github.com/user-attachments/assets/8b977fa5-2fbf-45a5-834b-22804d436864)

this is the output from chrome developer tool:
![image](https://github.com/user-attachments/assets/5c5c5e26-8eca-481c-afe6-a7776581abf3)

The only difference i can see between the two requests plan-core (that is 200 ok) and plan-management (404 not found) is the path for plan-core it is: 
/2d226d93-3e4e-4e4f-bb80-3d530b974e4b/_apis/build/builds/437831/73d43d90-87ad-4e26-83b3-0ff0dad7fdfd/cd72e36f-16af-57a0-4205-276a3b255da2/attachments/terraform-plan-results/plan-core

for plan-management it is:
/2d226d93-3e4e-4e4f-bb80-3d530b974e4b/_apis/build/builds/437831/73d43d90-87ad-4e26-83b3-0ff0dad7fdfd/ddf12597-4c66-5c23-18fd-838ff18747c6/attachments/terraform-plan-results/plan-management

**Comment by BHellstream** [comment]
if i look at a build that publish the two plans when i don't have changes in the parallel management stage the paths look like this:
plan-core: 
/2d226d93-3e4e-4e4f-bb80-3d530b974e4b/_apis/build/builds/437830/f5518139-e980-4433-a32b-fb9adbcbdd20/cd72e36f-16af-57a0-4205-276a3b255da2/attachments/terraform-plan-results/plan-core

plan-connectivity:
/2d226d93-3e4e-4e4f-bb80-3d530b974e4b/_apis/build/builds/437830/f5518139-e980-4433-a32b-fb9adbcbdd20/ce172455-0b98-5c15-7e61-bcce3da4a143/attachments/terraform-plan-results/plan-connectivity

plan-management:
/2d226d93-3e4e-4e4f-bb80-3d530b974e4b/_apis/build/builds/437830/f5518139-e980-4433-a32b-fb9adbcbdd20/ddf12597-4c66-5c23-18fd-838ff18747c6/attachments/terraform-plan-results/plan-management

so its the same difference in paths there, but it works.

**Comment by tomas0620** [comment]
Everything works for me. :) BUT—I am using CLASSIC pipelines, no YAML, so I have no multiple stages. I have only 2 plan tasks within my build pipeline.

**Comment by jason-johnson** [comment]
Ok, so it sounds like there may be some issue with parallel attachments but if so, that would be an issue with the AZDO library or Azure Devops itself.  My tasks just use the API to upload the attachments.  I would propose we close this issue, as it appears to be solved and we could open a separate issue on the concurrent plans to track it.  Any objections?
